history/subscriber/persister.py

```diff
@@ -268,6 +268,35 @@ def on_put(req, resp):
                 'Logging level must be DEBUG, INFO, WARNING, ERROR or CRITICAL!', 'level')
 
 
+def handle_choose_service(config, persister, check):
```

Comment:
The logic seems a little weird. My suggestion:

```
def start_dojot_messenger(config, persister, notifications_only):
    messenger = Messenger("Persister", config)    
    messenger.init()
    # TODO: add notifications to config on dojot-module-python
    messenger.create_channel("dojot.notifications", "r")
    messenger.on(config.dojot['subjects']['tenancy'], "message", persister.handle_new_tenant)        
    messenger.on("dojot.notifications", "message",   persister.handle_notification)
    LOGGER.info("Persisting notification events")
    if notifications_only != True: 
        messenger.create_channel(config.dojot['subjects']['devices'], "r")    
        messenger.create_channel(config.dojot['subjects']['device_data'], "r")
        messenger.on(config.dojot['subjects']['devices'], "message", persister.handle_event_devices)        
        messenger.on(config.dojot['subjects']['device_data'], "message", persister.handle_event_data)
        LOGGER.info("Persisting device events")

````

Reply:
Have you already given the solution? haha